### DISCUSSION TITLE
setValue without rerender

### DISCUSSION THREAD

**Discussion by edward1127** [react-hook-form/react-hook-form#10885] [Q&A] [answered]
I have a quite large form, some of the fields' values are set with 
```
      setValue(questionKey, dropItem.value, {
          shouldDirty: true,
          shouldTouch: true,
          shouldValidate: true,
        });
```
Based on the documentation 
![image](https://github.com/react-hook-form/react-hook-form/assets/45161984/813589f2-1c65-4f14-9401-5b4b143a5905)

The shouldTouch and shouldDirty will trigger re-render. Is there any way I can avoid that?

**Comment by Moshyfawn** [comment, marked as answer]
You can't avoid a re-render if you have an active `formState` subscription in your components tree; otherwise you'd get a stale value.

There's [a `disabled` prop for `useFormState`](https://react-hook-form.com/docs/useformstate) you can utilize to toggle your `formState` subscription l

**Reply by edward1127** [reply]
thank you for pointing the direction!
